### PR TITLE
feat: core package for seamless integration with amplitude analytics

### PR DESCRIPTION
**Pull request by bgiori**
### Summary

Add core package for integration with analytics SDK.

 * Keeps the same API as Android/iOS for `AnalyticsConnector` and `IdentityStore`
      * Changes the core singleton access to global/window in order to play nicely in the browser with potentially different script tags.
 * Package is pulled in as a dependency to experiment SDK and a dev dependency to the analytics SDK.

### Checklist

* [X] Does your PR title have the correct [title format](https://github.com/amplitude/experiment-js-client/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:   no

### CODE REVIEW

**Comment by bgiori** [comment]
This is not quite done yet, refactoring the component architecture.

File: packages/core/src/amplitudeCore.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { safeGlobal } from './util/global';
+
+safeGlobal['amplitudeCoreInstances'] = {};
+
+export class AmplitudeCore {
+  public readonly identityStore = new IdentityStoreImpl();
+  public readonly analyticsConnector = new AnalyticsConnectorImpl();
+
+  static getInstance(instanceName: string): AmplitudeCore {
+    if (!safeGlobal['amplitudeCoreInstances'][instanceName]) {
```
Comment (qingzhuozhen) [inline packages/core/src/amplitudeCore.ts:12]: Is this to resolve earlier issue mentioned ?

File: packages/core/rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import tsConfig from './tsconfig.json';
+
+const browserConfig = {
+  input: 'src/index.ts',
+  output: {
+    dir: 'dist',
+    entryFileNames: 'core.umd.js',
+    exports: 'named',
+    format: 'umd',
```
Comment (qingzhuozhen) [inline packages/core/rollup.config.js:16]: Curious is esm also supported?

File: packages/core/src/analyticsConnector.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  setEventReceiver(listener: AnalyticsEventReceiver): void;
+}
+
+export class AnalyticsConnectorImpl implements AnalyticsConnector {
+  private receiver: AnalyticsEventReceiver;
+  private queue: AnalyticsEvent[] = [];
+
+  logEvent(event: AnalyticsEvent): void {
+    if (!this.receiver) {
+      this.queue.push(event);
```
Comment (justin-fiedler) [inline packages/core/src/analyticsConnector.ts:20]: The queue could grow unbounded if no receiver is added. Maybe we could add a max queue size or user warning is no receiver is set.

Comment (justin-fiedler) [inline packages/core/src/analyticsConnector.ts:20]: The queue could grow unbounded if no receiver is added. Maybe we could add a max queue size or user warning is no receiver is set.

File: packages/core/src/amplitudeCore.ts (hunk 1 above)
Comment (bgiori) [reply packages/core/src/amplitudeCore.ts:12]: Yes, even if the same code is copied and used separately, the core instance is always a truly global singleton.

File: packages/core/rollup.config.js (hunk 2 above)
Comment (bgiori) [reply packages/core/rollup.config.js:16]: I need to add support for this later. High priority ATM but not blocking this release.

**Pull request merged**